mbed-client-cli/ns_cmdline.h

```diff
@@ -76,6 +76,15 @@ extern "C" {
 #define CMDLINE_RETCODE_COMMAND_CB_MISSING      -4  //!< Command callback function missing
 #define CMDLINE_RETCODE_COMMAND_NOT_FOUND       -5  //!< Command not found
 
+
+// #define MBED_CMDLINE_MINIMUM_CONFIG 0
+
+
+#ifndef MBED_CMDLINE_MINIMUM_CONFIG
```

Comment:
what could be better flag name ?

Reply:
> Also, instead of hardcoding the different buffer sizes and such for each feature set, they could be set in the mbed_lib.json, allowing easily overriding them in the application's mbed_app.json. 

those minimized buffer sizes are still "defaults" and can be overwrite in compiling time. It just reduce required options to be used when "minimized" version is wanted. mbed_lib.json is suitable only for mbed-os builds - not for example linux builds..

Reply:
> those minimized buffer sizes are still "defaults" and can be overwrite in compiling time. It just reduce required options to be used when "minimized" version is wanted.

True.

> mbed_lib.json is suitable only for mbed-os builds - not for example linux builds..

That's also true. Thinking about it now, if it's only a matter of enabling or disabling the minimum configuration, the flagging as you did it is fine. Could perhaps add a verb to make it explicit what setting the flag does, i.e. `MBED_CMDLINE_USE_MINIMUM_CONFIG` might be clearer than just `MBED_CMDLINE_MINIMUM_CONFIG`.

Reply:
`MBED_CMDLINE_USE_MINIMUM_CONFIG` sounds better name, I'll change to it.